src/sources/postgres.py

```diff
@@ -31,8 +33,10 @@ class PostgresSource(Source[DataFrame]):
     """
 
     def __init__(self, db_url: str, query_string: str):
-        self.query_string = query_string
         self.engine: sqlalchemy.engine.Engine = create_engine(db_url)
+        self.query_string = ""
+        self._set_query_string(query_string)
+        self.validate()
```

Comment:
Lets do this:

https://github.com/bh2smith/dune-sync/pull/50#discussion_r1832107124